### PR TITLE
SECOND ENTITY AND MANY-TO-MANY ENTITY RELATIONSHIPS

### PR DESCRIPTION
**Pull request by Nadya1993**
https://www.getpostman.com/collections/6211e69d33fdedede493

### CODE REVIEW

File: src/services/GroupService.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  getGroups = async (): Promise<GroupAttributes[]> => (
+    await this.groupModel.findAll({
+      attributes: ['name', 'permission', 'group_id'],
+      include: UserModel
+    })
+  )
+
+  findGroup = async (id: string): Promise<GroupAttributes | undefined> => (
+    await (await this.groupModel.findByPk(id, {
```
Comment (spiderpoul) [inline src/services/GroupService.ts]: i suggest splitting it for more readability

File: src/services/GroupService.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      attributes: ['name', 'permission', 'group_id']
+    }))
+  )
+
+  addGroup = async (name: string, permission: Permission[]): Promise<GroupInstance> => (
+    await this.groupModel.create({ groupId: uuidv4(), name, permission })
```
Comment (spiderpoul) [inline src/services/GroupService.ts:28]: useless `async` `await`

File: src/routers/groupRoutes.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+const router = express.Router();
+
+// find all groups
+router.get('/', async (req: express.Request, res: express.Response) => {
+  res.json(await GroupServiceInstance.getGroups());
+});
+
+// delete group by id
+router.post('/', async (req: express.Request, res: express.Response) => {
```
Comment (spiderpoul) [inline src/routers/groupRoutes.ts]: why not using `delete` method instead of `post`?

File: src/services/GroupService.ts (hunk 2 above)
Comment (Nadya1993) [reply src/services/GroupService.ts:28]: All sequelize methods, that manage database data, are asynchronous according to specification: https://sequelize.org/master/manual/model-querying-basics.html#simple-insert-queries

File: src/services/GroupService.ts (hunk 1 above)
Comment (Nadya1993) [reply src/services/GroupService.ts]: get rid of extra await

File: src/routers/groupRoutes.ts (hunk 3 above)
Comment (Nadya1993) [reply src/routers/groupRoutes.ts]: Corrected router👍

**Pull request merged**